libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -67,18 +67,13 @@ class barrier : public _CUDA_VSTD::__barrier_base<_CompletionF, _Sco> {
 
     _LIBCUDACXX_INLINE_VISIBILITY
     friend void init(barrier * __b, _CUDA_VSTD::ptrdiff_t __expected) {
-#if (_LIBCUDACXX_DEBUG_LEVEL >= 2)
-        _LIBCUDACXX_DEBUG_ASSERT(__expected >= 0);
-#endif
-
+        _LIBCUDACXX_DEBUG_ASSERT(__expected >= 0, "Cannot initialize barrier with negative arrival count");
```

Comment:
Does this expand to no code if `_LIBCUDACXX_DEBUG_LEVEL < 2` ?

Reply:
This is an assert that we use in LIBCUDACXX

It expands to nothing unless  one of the following is defined:

- `_LIBCUDACXX_DEBUG`
- `_LIBCUDACXX_ENABLE_DEBUG_MODE`
- `_LIBCUDACXX_ENABLE_ASSERTIONS`